lib/datadog/core/configuration/settings.rb

```diff
@@ -490,7 +472,7 @@ def initialize(*_)
         # @default `->{ Time.now }`
         # @return [Proc<Time>]
         option :time_now_provider do |o|
-          o.default { ::Time.now }
+          o.default { -> { ::Time.now } }
```

Comment:
This is a weird use case. We want the default value to be a block, so we need to have a block that returns a block. 

Before we did not mark the settings as `lazy`, so rather than execute the block, we store the block as the default value. To achieve the same thing, we have to wrap the block into a block.